tests/src/test/scala/whisk/core/containerpool/test/ContainerArgsConfigTest.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.containerpool.test
+
+import org.junit.runner.RunWith
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.junit.JUnitRunner
+import pureconfig._
+import whisk.core.ConfigKeys
+import whisk.core.containerpool.ContainerArgsConfig
+
+@RunWith(classOf[JUnitRunner])
+class ContainerArgsConfigTest extends FlatSpec with Matchers {
+
+  it should "use defaults for container args map" in {
+    val config = loadConfigOrThrow[ContainerArgsConfig](ConfigKeys.containerArgs)
+
+    //check defaults
+    config.network shouldBe "bridge"
+    config.dnsServers shouldBe Seq[String]()
+    config.extraArgs shouldBe Map[String, Set[String]]()
+  }
+
+  it should "override defaults from system properties" in {
+    System.setProperty("whisk.container-factory.container-args.extra-args.label.0", "l1")
```

Comment:
May be the tests should revert changes in System property post run otherwise it may lead to side effects in other test

Reply:
agreed - since this just got merged, I will address in a future pr - this area will get some revisions as I work on some of the containerfactory details for mesos